src/components/AST/Viz.js

```diff
@@ -0,0 +1,192 @@
+import React, { Fragment, useCallback, useMemo, useState } from "react";
+import { parse } from "@babel/parser";
+import { Accordion, Grid, Menu, Popup, Segment } from "semantic-ui-react";
+import UglyPopup, { lookUpNodeType } from "../Popup";
+
+// DFS through AST generated by Babel parser
+
+function CompositeObj({ k, obj, cursor, setPos }) {
+  // setPos is memoized so dependency is just to placate warnings.
+  const content = useMemo(() => {
+    return Object.entries(obj).map(([i, v], index) => {
+      return (
+        <Fragment key={index}>
+          {typeof v === "object" && v !== null ? (
+            <Composite k={i} ast={v} cursor={cursor} setPos={setPos} />
+          ) : (
+            <Primitive k={i} val={v} />
```

Comment:
What are `k`, `i`, `v` shorthand for?

Reply:
`k`, short for `key`, which is already being used in the AST, is any sort of label of what that part of the AST is. `i`, `v` are key and value, respectively, of the entries of the AST in that closure only.

Reply:
I'll allow it. `k` `v` and `i` are pretty common shorthand.